### ISSUE TITLE
findOne() with @EntityGraph on it does not return all @OneToMany elements

### ISSUE THREAD

**Issue by pisarek** [spring-projects/spring-data-jpa#2914]
spring-boot-starter-data-jpa:3.0.4

TLDR: Using `findOne()` annotated with `@EntityGraph` results in not all rows from child relationship being returned.

Given the following model:

```java
@Entity
@Getter
@Setter
@Table(name = "A")
public class A {
    @Id
    @Column(name = "ref_id")
    String refId;

    @OneToMany(mappedBy = "a")
    private List<B> bs = new ArrayList<>();
}

@Entity
@Getter
@Setter
@Table(name = "B")
@IdClass(BID.class)
public class B {

    @Id
    String refId;

    @Id
    String bId;

    @ManyToOne(optional = false)
    @JoinColumn(name = "ref_id", referencedColumnName = "ref_id", insertable = false, updatable = false)
    private A a;
}

@Getter
@Setter
public class BID {

    @Column(name = "ref_id")
    String refId;

    @Column(name = "b_id")
    String bId;
}

``` 

when I run it using `Optional<A> findOne(Specification<A> spec)` like so:
`aRepository.findOne((root, query, criteriaBuilder) -> criteriaBuilder.equal(root.get("refId"), "12"));`
I get all the Bs. But there are two queries to the DB.

```
Hibernate: 
    select
        a1_0.ref_id 
    from
        a a1_0 
    where
        a1_0.ref_id=? fetch first ? rows only
Hibernate: 
    select
        b1_0.ref_id,
        b1_0.b_id 
    from
        b b1_0 
    where
        b1_0.ref_id=?
```
 So I tried to solve it by putting `@EntityGraph` on `findOne()` like so:
```java
@Override
@EntityGraph(attributePaths = "bs")
Optional<A> findOne(Specification<A> spec);
``` 
and it works, there is only one query:

```
select
        a1_0.ref_id,
        b1_0.ref_id,
        b1_0.b_id 
    from
        a a1_0 
    left join
        b b1_0 
            on a1_0.ref_id=b1_0.ref_id 
    where
        a1_0.ref_id=? fetch first ? rows only
```

But unfortunately, **only 2 of the Bs from the relationship are returned**, regardless of how many there are in the DB.

**Comment by schauder** [comment]
This sounds very much like a Hibernate issue, since all the JPA mapping is done by it.
I recommend creating a reproducer based purely on Hibernate and submitting an issue with them.

If you think this is actually a Spring Data issue, please provide a reproducer for the issue including a test case that demonstrates the correct behaviour using the `EntityManager` directly.
